tensorflow/core/kernels/data/map_dataset_op_test.cc

```diff
@@ -0,0 +1,363 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <tuple>
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/framework/fake_input.h"
+#include "tensorflow/core/framework/function.h"
+#include "tensorflow/core/framework/function_handle_cache.h"
+#include "tensorflow/core/framework/function_testlib.h"
+#include "tensorflow/core/framework/node_def_builder.h"
+#include "tensorflow/core/framework/partial_tensor_shape.h"
+#include "tensorflow/core/framework/variant.h"
+#include "tensorflow/core/framework/variant_tensor_data.h"
+#include "tensorflow/core/kernels/data/dataset_test_base.h"
+#include "tensorflow/core/kernels/data/dataset_utils.h"
+#include "tensorflow/core/kernels/data/iterator_ops.h"
+#include "tensorflow/core/kernels/ops_testutil.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/util/ptr_util.h"
+
+namespace tensorflow {
+namespace data {
+namespace {
+
+class MapDatasetOpTest : public DatasetOpsTestBase {
+ public:
+  static const char* kNodeName;
+  static const char* kOpName;
+
+  ~MapDatasetOpTest() override {
+    gtl::STLDeleteElements(&tensors_);
+    range_dataset_->Unref();
+  }
+
+ protected:
+  // Creates a new MapDataset dataset, meanwhile initializing the MapDataset op
+  // kernel and context internally.
+  DatasetBase* CreateMapDataset(DatasetBase* input_dataset,
```

Comment:
for consistency, change the signature to return `Status`, replacing TF_CHECK_OK with TF_RETURN_IF_ERROR

Reply:
also, the created dataset should be returned through an output argument and not stored in the instance